backend/routes/run.js

```diff
@@ -0,0 +1,194 @@
+import { PrismaClient } from "@prisma/client";
+import {
+  STAT_MEANS,
+  STAT_VARIANCES,
+  POP_SIZE,
+  incrementPopSize,
+} from "./statDictionaries.js";
+const prisma = new PrismaClient();
+
+async function updatePlayer(player) {
+  const oldPlayer = await prisma.player.findUnique({
+    where: { id: player.id },
+  });
+
+  if (oldPlayer == null) {
+    player = await createPlayer(player);
+  } else {
+    player = await changePlayerData(player, oldPlayer);
+  }
```

Comment:
These two functions are basically duplicated. How can we reduce the amount of copy-paste here and abstract the common logic?

Reply:
Taking the processing of percentage values up one level, and using the prisma upsert method to update a record or create a new one if it does not exist.

Reply:
Using upsert still has a signifcant amount of repeated code for passing parameters through the update and create fields. Will revisit tomorrow.